src/java.naming/share/classes/com/sun/jndi/ldap/pool/PooledConnectionFactory.java

```diff
@@ -44,4 +44,13 @@
      */
     public abstract PooledConnection createPooledConnection(PoolCallback pcb)
         throws NamingException;
+
+    /**
+     * Creates a pooled connection.
+     * @param pcb callback responsible for removing and releasing the pooled
+     * connection from the pool.
+     * @param timeout the connection timeout
+     */
+    public abstract PooledConnection createPooledConnection(PoolCallback pcb, long timeout)
```

Comment:
why not use int timeout to be consistent with existing code ?
You've been required to "squash" it into an int in the factory ?

Reply:
Well I guess the request was "why not use long everywhere to avoid casting to int" ;-)
But I'm happy with either too - as long as the place where you have a long (e.g obtained by substracting two nano times) and call a method that takes an int has the proper guards in place, and either assert/throws/floor or ceil if the assumptions are not met - provided that a comment explains why that particular alternative is selected.

Reply:
Good point. I think its better to deal with the casts at the edges since the timeout handling will use long by default.

Reply:
yes a redeclaration of  timeout with a type long  across the  component would be a consistent approach, also

Reply:
Not in this issue. I plan to file a follow up bug to make a slight change to the test. I would like to get this issue fixed ASAP and would appreciate the time to take a good look at the transition to a long timeout.